WcaOnRails/app/views/competitions/_registration_requirements.html.erb

```diff
@@ -0,0 +1,64 @@
+<% if @competition.use_wca_registration? %>
+  <%= t('competitions.competition_info.create_wca_account_html', here: link_to(t('common.here'), new_user_registration_path)) %>
+  <br>
+  <%= t('competitions.competition_info.claim_wca_id_html', here: link_to(t('common.here'), profile_claim_wca_id_path)) %>
+  <br>
+  <%= t('competitions.competition_info.register_html', here: link_to(t('common.here'), competition_register_path(@competition))) %>
+  <br>
+<% elsif @competition.website %>
+  <%= t('competitions.competition_info.register_html', here: link_to(t('common.here'), @competition.website, target: "_blank")) %>
+  <br>
+<% else %>
+<% end %>
+<% if @competition.competitor_limit_enabled? %>
+  <%= t('competitions.competition_info.competitor_limit', competitor_limit: @competition.competitor_limit) %>
+  <br>
+<% elsif !@competition.competitor_limit_enabled? %>
+  <%= t('competitions.competition_info.no_competitor_limit') %>
+  <br>
+<% else %>
+<% end %>
+<% if @competition.has_fees? %>
+  <%= t('competitions.competition_info.entry_fee', base_entry_fee: format_money(@competition.base_entry_fee)) %>
+  <br>
+  <% @competition.competition_events.each do |competition_event| %>
+    <% if competition_event.has_fee? %>
+      <dt><%= competition_event.event.name %></dt>
+      <dd><%= format_money competition_event.fee %></dd>
+    <% end %>
+  <% end %>
+<% elsif !@competition.has_fees? %>
+  <%= t('competitions.competition_info.no_entry_fee') %>
+  <br>
+<% else %>
+<% end %>
+<% if @competition.refund_policy_percent? %>
+  <%= t('competitions.competition_info.refund_policy_html', refund_policy_percent: number_to_percentage(@competition.refund_policy_percent, precision: 0), limit_date_and_time: wca_local_time(@competition.refund_policy_limit_date)) %>
+  <br>
+<% elsif !@competition.refund_policy_percent? %>
+  <%= t('competitions.competition_info.no_refunds') %>
+  <br>
+<% else %>
+<% end %>
+<% if @competition.on_the_spot_registration? %>
+  <% if @competition.on_the_spot_entry_fee_lowest_denomination? %>
+    <%= t('competitions.competition_info.on_the_spot_registration', on_the_spot_base_entry_fee: format_money(@competition.on_the_spot_base_entry_fee)) %>
+    <br>
+  <% else %>
+    <%= t('competitions.competition_info.on_the_spot_registration_free') %>
+    <br>
+  <% end %>
+<% elsif !@competition.on_the_spot_registration? %>
+  <%= t('competitions.competition_info.no_on_the_spot_registration') %>
+  <br>
+<% else %>
+<% end %>
+<% if @competition.guests_entry_fee? %>
+  <%= t('competitions.competition_info.guests_pay', guests_base_fee: format_money(@competition.guests_base_fee)) %>
+  <br>
+<% elsif !@competition.guests_entry_fee? %>
```

Comment:
Are you sure all these `if...elsif..end` statements you wrote in this file do what you want? They look like this to me:

```ruby
if foo
  ....
elsif !foo
  ....
else
end
```

It seems to me like it's going to be impossible for the "else, don't render anything" block to happen here. Won't we always go into the `if`, or the `elsif`?

I would expect the check to look more like this:

```ruby
if foo.nil?
   # we don't know the value of "foo", so probably render nothing
elsif something?
   ...
else
   ...
end
```

Reply:
I think it does what I want from looking at competitions and creating new ones on local, but I also think that your proposal is easier to follow, so I'll change this.